playbooks/roles/start/tasks/wait_for_zowe.yml

```diff
@@ -42,21 +43,24 @@
 
 # ============================================================================
 # Wait for Zowe services be started
-- name: Wait for Desktop port {{ zowe_zlux_port }} to be available
+- name: Wait for Desktop at {{zowe_zos_host}} port {{ zowe_zlux_port }} to be available
   uri:
-    url: "https://{{ zowe_external_domain_name }}:{{ zowe_zlux_port }}"
+    url: "https://{{ zowe_zos_host }}:{{ zowe_zlux_port }}"
     follow_redirects: all
     method: GET
     validate_certs: false
   register: _result
   until: _result.status == 200
   retries: "{{ wait_for_zowe_service_retries | int }}"
   delay: "{{ wait_for_zowe_service_delay | int }}"
+  # if zowe_zos_host and zowe_external_domain_name are not same, zlux could be
+  # started off z/OS and then we couldn't test right now
+  when: zowe_zos_host == zowe_external_domain_name and "app-server" in zowe_launch_components
```

Comment:
Please change to `when: zowe_zos_host == zowe_external_domain_name and ("app-server" in zowe_launch_components or zowe_launch_components == "")`